modules/wallet/defrag_test.go

```diff
@@ -198,3 +199,45 @@ func TestDefragOutputExhaustion(t *testing.T) {
 	close(closechan)
 	<-donechan
 }
+
+type mockTpool struct {
+	minFee types.Currency
+	maxFee types.Currency
+}
+
+func (t *mockTpool) AcceptTransactionSet(_ []types.Transaction) error  { return nil }
+func (t *mockTpool) Broadcast(_ []types.Transaction)                   {}
+func (t *mockTpool) Close() error                                      { return nil }
+func (t *mockTpool) FeeEstimation() (types.Currency, types.Currency)   { return t.minFee, t.maxFee }
+func (t *mockTpool) ProcessConsensusChange(cc modules.ConsensusChange) {}
+func (t *mockTpool) PurgeTransactionPool()                             {}
+func (t *mockTpool) Transaction(id types.TransactionID) (types.Transaction, []types.Transaction, bool) {
+	return types.Transaction{}, []types.Transaction{}, true
+}
+func (t *mockTpool) TransactionList() []types.Transaction                                  { return []types.Transaction{} }
+func (t *mockTpool) TransactionPoolSubscribe(subscriber modules.TransactionPoolSubscriber) {}
+func (t *mockTpool) Unsubscribe(subscriber modules.TransactionPoolSubscriber)              {}
```

Comment:
You should be able to use a real tpool as the base for your mock tpool, right?  Then you just overwrite the FeeEstimation function, and it should be a lot cleaner overall.